src/CodeGen_ARM.cpp

```diff
@@ -705,16 +705,43 @@ void CodeGen_ARM::visit(const Store *op) {
             args[i] = codegen(call->args[i]);
         }
 
-        // Declare the function
+#if LLVM_VERSION > 39
+        Value *shuffle_part1, *shuffle_part2;
```

Comment:
It is target independent, yes. Yet, this pattern is currently recognized by an LLVM pass and lowered to (V)STn intrinsics just for ARM and AARch64. It may be costly to have all these shuffles for all archs. Note also that the previous check about num_vecs of 2,3,or 4 is still there.
I haven't tested yet how the additional shuffles in this patch lower for x86 for example.
Let me look in more detail at Stores in CodeGen_LLVM.